src/services/performanceDbApi.js

```diff
@@ -86,19 +86,10 @@ ORDER BY
 
     const result = await datasette.runQuery(query)
 
-    // convert the rows into an easier to access format
-    const columns = result.columns
-    const rows = result.rows.map((row) => {
-      return row.reduce((acc, val, index) => {
-        acc[columns[index]] = val
-        return acc
-      }, {})
-    })
-
-    const datasets = rows.reduce((accumulator, row) => {
+    const datasets = result.formattedData.reduce((accumulator, row) => {
       let error
-      if (row.http_status !== '200' || row.exception !== '') {
-        error = row.exception !== '' ? row.exception : `endpoint returned with a status of ${row.http_status}`
+      if ((row.http_status !== '200' && row.http_status !== 200) || row.exception) {
```

Comment:
This check for 'string or number' seems to be here only because the mock data in `performanceDbApi.test.js` incorrectly passes `http_status` as number (the SQL query indicates that the actual value will always be a string). So we end up with production code dealing with broken mock data, which we should obviously avoid.

Reply:
good spot, fixed this